canonical/dex-auth-operator
# feat: add dex_oidc_config library

Add a library to handle the dex-oidc-config relation interface. This library allows dex-auth to share its OIDC configuration with OIDC clients such as the oidc-gatekeeper charm. The library is designed to be extended to share any information about Dex, but right now it is only sharing Dex's issuer url.

Closes #203

#### Testing instructions
1. Deploy the charm from this PR
2. Deploy the charm from https://github.com/canonical/oidc-gatekeeper-operator/pull/163
3. Wait for them to become active and idle
4. Add the relation between them (interface dex-oidc-config)
5. Verify that the requirer is actually getting data from dex-auth `juju show-unit oidc-gatekeeper`. There should be the `dex-issuer-url` value.
